OpenSim/Common/Exception.h

```diff
@@ -67,14 +110,31 @@ class OSIMCOMMON_API Exception  : public std::exception {
     std::string _file;
     /** Line number at which the error occurred. */
     int _line;
+    /** Delimiter/separator used to separate different parts of the message.  */
+    static const std::string _msgSeparator;
```

Comment:
I don't think I understand.
>I disagree. If we don't have a use for it now it adds complexity for no gain.

What exactly is complex about this constant ? 
The gain here is to be able to adjust the delimiter if desired. 
How complicated is it to change the formatting in future (to one of the things you mention) if we have this constant which is currently being used in one function only (addMessage())?